File: SolariX.Api/DTOs/AuthDtos.cs
// ============================================================================
// File: AuthDtos.cs
// Project: SolariX - Smart Solar Microgrid Trading System
// Description: Data transfer objects for user authentication, registration, and profile management.
// Module: SE4040 Enterprise Application Development
// ============================================================================

using System.ComponentModel.DataAnnotations;
using SolariX.Api.Models;

namespace SolariX.Api.DTOs
{
    public class LoginRequest
    {
        [Required]
        public string Identifier { get; set; } = string.Empty; // NIC or Email

        [Required]
        public string Password { get; set; } = string.Empty;
    }

    public class LoginResponse
    {
        public string NIC { get; set; } = string.Empty;
        public string FullName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Role { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public string Token { get; set; } = string.Empty;
    }

    public class RegisterProsumerRequest
    {
        [Required]
        public string NIC { get; set; } = string.Empty;

        [Required]
        public string FullName { get; set; } = string.Empty;

        [Required]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        [Required]
        public string PhoneNumber { get; set; } = string.Empty;

        [Required]
        [MinLength(6)]
        public string Password { get; set; } = string.Empty;

        public string Address { get; set; } = string.Empty;

        public double SolarCapacityKW { get; set; } = 5.0;
    }

    public class CreateOperatorRequest
    {
        [Required]
        public string NIC { get; set; } = string.Empty;

        [Required]
        public string FullName { get; set; } = string.Empty;

        [Required]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        [Required]
        public string PhoneNumber { get; set; } = string.Empty;

        [Required]
        [MinLength(6)]
        public string Password { get; set; } = string.Empty;

        [Required]
        public UserRole Role { get; set; } = UserRole.GridOperator;
    }

    public class UpdateProfileRequest
    {
        public string FullName { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public double SolarCapacityKW { get; set; }
    }

    public class UserResponse
    {
        public string Id { get; set; } = string.Empty;
        public string NIC { get; set; } = string.Empty;
        public string FullName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public string Role { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public double SolarCapacityKW { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: mobile-client/app/src/main/java/com/solarix/mobile/data/remote/dtos/AuthDtos.kt
package com.solarix.mobile.data.remote.dtos

import com.google.gson.annotations.SerializedName

data class LoginRequest(
    @SerializedName("identifier") val identifier: String,
    @SerializedName("password") val password: String
)

data class LoginResponse(
    @SerializedName("nic") val nic: String,
    @SerializedName("fullName") val fullName: String,
    @SerializedName("email") val email: String,
    @SerializedName("role") val role: String,
    @SerializedName("status") val status: String,
    @SerializedName("token") val token: String
)

data class RegisterProsumerRequest(
    @SerializedName("nic") val nic: String,
    @SerializedName("fullName") val fullName: String,
    @SerializedName("email") val email: String,
    @SerializedName("phoneNumber") val phoneNumber: String,
    @SerializedName("password") val password: String,
    @SerializedName("address") val address: String,
    @SerializedName("solarCapacityKW") val solarCapacityKW: Double = 5.0
)

data class UpdateProfileRequest(
    @SerializedName("fullName") val fullName: String,
    @SerializedName("phoneNumber") val phoneNumber: String,
    @SerializedName("address") val address: String,
    @SerializedName("solarCapacityKW") val solarCapacityKW: Double
)

data class UserResponse(
    @SerializedName("id") val id: String,
    @SerializedName("nic") val nic: String,
    @SerializedName("fullName") val fullName: String,
    @SerializedName("email") val email: String,
    @SerializedName("phoneNumber") val phoneNumber: String,
    @SerializedName("role") val role: String,
    @SerializedName("status") val status: String,
    @SerializedName("address") val address: String,
    @SerializedName("solarCapacityKW") val solarCapacityKW: Double,
    @SerializedName("createdAt") val createdAt: String?
)

data class ApiResponse<T>(
    @SerializedName("message") val message: String? = null,
    @SerializedName("error") val error: String? = null,
    @SerializedName("data") val data: T? = null
)
